contracts/TxRelay.sol

```diff
@@ -0,0 +1,60 @@
+pragma solidity 0.4.8;
+
+//This contract is meant as a "singleton" forwarding contract.
+//Eventually, it will be able to forward any transaction to
+//Any contract that is built to accept it.
+
+contract TxRelay {
+
+  // Note: This is a local nonce.
+  // Different from the nonce defined w/in protocol.
+  mapping(address => uint) nonce;
+
+  /*
+   * @dev Relays meta transactions
+   * @param sigV, sigR, sigS ECDSA signature on some data to be forwarded
+   * @param data The bytes necessary to call the function in the destination contract.
+                 Note, the first encoded argument in data must be address of the signer
+   * @param claimedSender Address of the user who is having tx forwarded
+   */
+  function relayMetaTx(uint8 sigV, bytes32 sigR, bytes32 sigS,
+                       address destination, bytes data,
+                       address claimedSender) {
```

Comment:
Now that I think about it though, it should just be a single line of assembly and would remove:
68 gas per non-zero char in input. 4 gas per zero char in input. An address is 20 bytes, so 12 * 4 + 20 (1/10) * 4 + 20 (9/10) * 68 = 1,280.

Thus, including this duplicated data is currently adding an expected 1,280 gas to every call - so we should do this if we can do it for less than that. I will do some experiments and report back :)